internal/daemonset/worker.go

```diff
@@ -0,0 +1,246 @@
+				NodeSelectorTerms: obj,
+			},
+		},
+	}
+
+}
+
+func BoolPointer(b bool) *bool {
+	return &b
+}
+
+func getWorkerSecurityContext() *corev1.SecurityContext {
+	return &corev1.SecurityContext{
+		ReadOnlyRootFilesystem:   BoolPointer(true),
+		RunAsNonRoot:             BoolPointer(true),
+		AllowPrivilegeEscalation: BoolPointer(false),
+		SeccompProfile: &corev1.SeccompProfile{
+			Type: "RuntimeDefault",
+		},
+		Capabilities: &corev1.Capabilities{
+			Drop: []corev1.Capability{"ALL"},
+		},
+	}
+}
+
+func getWorkerVolumeMounts() *[]corev1.VolumeMount {
+
+	containerVolumeMounts := []corev1.VolumeMount{
+		{
+			Name:      "host-boot",
+			MountPath: "/host-boot",
+			ReadOnly:  true,
+		},
+		{
+			Name:      "host-os-release",
+			MountPath: "/host-etc/os-release",
+			ReadOnly:  true,
+		},
+		{
+			Name:      "host-sys",
+			MountPath: "/host-sys",
+		},
+		{
+			Name:      "nfd-worker-config",
+			MountPath: "/etc/kubernetes/node-feature-discovery",
+		},
+		{
+			Name:      "nfd-hooks",
+			MountPath: "/etc/kubernetes/node-feature-discovery/source.d",
+		},
+		{
+			Name:      "nfd-features",
+			MountPath: "/etc/kubernetes/node-feature-discovery/features.d",
+		},
+	}
+
+	return &containerVolumeMounts
+}
+
+func getWorkerVolumes() []corev1.Volume {
+	containerVolume := []corev1.Volume{
+		{
+			Name: "host-dev",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/dev",
+				},
+			},
+		},
+		{
+			Name: "host-boot",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/boot",
+				},
+			},
+		},
+		{
+			Name: "host-os-release",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/etc/os-release",
+				},
+			},
+		},
+		{
+			Name: "host-sys",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/sys",
+				},
+			},
+		},
+		{
+			Name: "nfd-hooks",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/etc/kubernetes/node-feature-discovery/source.d",
+				},
+			},
+		},
+		{
+			Name: "nfd-features",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/etc/kubernetes/node-feature-discovery/features.d",
+				},
+			},
+		},
+		{
+			Name: "nfd-worker-config",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{Name: "nfd-worker"},
+					Items: []corev1.KeyToPath{
+						{
+							Key:  "nfd-worker-conf",
+							Path: "nfd-worker.conf",
+						},
+					},
+				},
+			},
+		},
+	}
+	return containerVolume
+}
+
+func getWorkerLabelsAForApp(name string) map[string]string {
+	return map[string]string{"app": name}
+}
+
+func (d *daemonset) SetWorkerDaemonsetAsDesired(ctx context.Context, nfdInstance *nfdv1.NodeFeatureDiscovery, workerDS *appsv1.DaemonSet) error {
```

Comment:
it seems a little strange that SetTopology is defined in daemonset.go and SetWorker is defined in the worker.go. I would suggest that both those function will be defined in the daemonset.go.  The helper functions (getLabels,getTolerations etc') can be defined in a dedicated file (worker.go). In addition we can keep the unit tests defined in the daemonset_test.go, without defining dedicated test file per each function